zebra-state/src/service/arbitrary.rs

```diff
@@ -97,7 +126,9 @@ impl Strategy for PreparedChain {
         }
 
         let chain = chain.clone().expect("should be generated");
-        let count = (1..chain.1.len()).new_tree(runner)?;
+        // `count` must be 1 less since the first block is used to build the
+        // history tree.
+        let count = (1..chain.1.len() - 1).new_tree(runner)?;
```

Comment:
Readability: Can we do the chain length adjustment in the same function that we use the first block to build the history tree?

Otherwise, the code is harder to understand.

We can either:
* adjust the count in every test, or
* build the history tree in this strategy and add it to the strategy's return type

Reply:
I've implemented the second option in 52710308
Indeed it's much more readable. (Even I was confused with my own code while improving this :sweat_smile:)

Reply:
Oops, that broke other tests. Fixed in 87de67a4 - it's not as neat, but I think it's a bit better than the original approach...